### PR TITLE
Fix escaping quotes in parameter description when export model as python

### PR DESCRIPTION
**Pull request by Koyaani**
## Description

Fix the issue #42718. When there is a single quote in a parameter description, the asPythonString method is broken. As @nicogodet suggested, I ensured that there was as less escaping as possible.

A solution is to create a method `descriptionAsPythonString` in `QgsProcessingParameterDefinition` that does the escaping before using it in `asPythonString` of child classes.

I don't know which version I should put for \since comment.

### CODE REVIEW

File: src/core/processing/qgsprocessingparameteraggregate.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ QString QgsProcessingParameterAggregate::asPythonString( QgsProcessing::PythonOu
   {
     case QgsProcessing::PythonQgsProcessingAlgorithmSubclass:
     {
-      QString code = QStringLiteral( "QgsProcessingParameterAggregate('%1', '%2'" ).arg( name(), description() );
+      QString code = QStringLiteral( "QgsProcessingParameterAggregate('%1', %2" ).arg( name(), descriptionAsPythonString() );
```
Comment (nyalldawson) [inline src/core/processing/qgsprocessingparameteraggregate.cpp:77]: There's an existing method "QgsProcessingUtils::stringToPythonLiteral" which could be used here instead, and avoid the need for the new descriptionAsPythonString() methd. (I.e. you could wrap description() in stringToPythonLiteral() and then implement your refined logic directly in stringToPythonLiteral

Comment (Koyaani) [reply src/core/processing/qgsprocessingparameteraggregate.cpp:77]: Your solution with `stringToPythonLiteral` is much better. However, with the logic I propose, this function will no longer escape all quotes (single and double). But I don't think it is necessary in Python to escape double quotes in strings defined by single quotes. And inversely.

To take a test : `uri='complex' username="complex"` become `'uri=\'complex\' username=\"complex\"'` and with my proposition become `'uri=\'complex\' username="complex"'`

Comment (nyalldawson) [reply src/core/processing/qgsprocessingparameteraggregate.cpp:77]: Could you move your refined logic to stringToPythonLiteral though? That'd benefit a whole lot of other places in Processing too then, not just descriptions.

Comment (Koyaani) [reply src/core/processing/qgsprocessingparameteraggregate.cpp]: With a little thought, I am afraid that the generated files will have an inconsistent mix of single and double quotes.

Comment (nyalldawson) [reply src/core/processing/qgsprocessingparameteraggregate.cpp]: I'm ok with that -- the new logic you've implemented matches the internal logic I use when writing python code. 👍

One nice potential follow up would be adding logic to switch to """.....""" format for very long strings, but that's just extra icing 😁

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**